lib/ansible/modules/monitoring/zabbix/zabbix_import.py

```diff
@@ -0,0 +1,260 @@
+
+DOCUMENTATION = '''
+---
+module: zabbix_import
+short_description: Import zabbix configuration from xml or json
+description:
+   - This module allows you to import zabbix configuration from xml or json file
+version_added: "2.7"
+author:
+    - "(@kuczko)"
+requirements:
+    - "python >= 2.6"
+    - zabbix-api
+options:
+    import_file:
+        description:
+            - Path to file with zabbix import
+        required: true
+    import_format:
+        description:
+            - Format of zabbix import file
+        choices: ['xml','json']
+        default: xml
+    rules:
+        description:
+            - "Rules for importing items. By default it will switch all rules to true (overwritting/removing all). It means:"
+            - "applications:"
+            - "  createMissing: true"
+            - "  deleteMissing true"
+            - "discoveryRules:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "  deleteMissing: true"
+            - "graphs:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "  deleteMissing: true"
+            - "groups:"
+            - "  createMissing: true"
+            - "hosts:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "images:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "items:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "  deleteMissing: true"
+            - "maps:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "screens:"
+            - "  createMissing: true"
+            - "  updateExisting': true"
+            - "templateLinkage:"
+            - "  createMissing: true"
+            - "templates:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "templateScreens:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "  deleteMissing: true"
+            - "triggers:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "  deleteMissing: true"
+            - "valueMaps:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+
+
+extends_documentation_fragment:
+    - zabbix
+'''
+
+EXAMPLES = '''
+- name: Import zabbix hosts
+  local_action:
+    module: zabbix_import
+    server_url: http://monitor.example.com
+    login_user: username
+    login_password: password
+    import_file: /tmp/myfile.xml
+    import_format: xml
+    rules:
+      hosts:
+        createMissing: True
+    timeout: 10
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    from zabbix_api import ZabbixAPI
+
+    HAS_ZABBIX_API = True
+except ImportError:
+    HAS_ZABBIX_API = False
+
+
+DEFAULT_RULES = ({
```

Comment:
These defaults work fine imo. I don't see why would someone want to manage zabbix hosts partially via XML/JSON imports and partially via web console until it is one-time import thingy.